doc/developers/performance.rst

```diff
@@ -325,37 +325,8 @@ standalone function in a ``.pyx`` file, add static type declarations and
 then use Cython to generate a C program suitable to be compiled as a
 Python extension module.
 
-The official documentation available at http://docs.cython.org/ contains
-a tutorial and reference guide for developing such a module. In the
-following we will just highlight a couple of tricks that we found
-important in practice on the existing cython codebase in the scikit-learn
-project.
-
-TODO: html report, type declarations, bound checks, division by zero checks,
-memory alignment, direct blas calls...
-
-- https://www.youtube.com/watch?v=gMvkiQ-gOW8
-- https://conference.scipy.org/proceedings/SciPy2009/paper_1/
-- https://conference.scipy.org/proceedings/SciPy2009/paper_2/
-
-Using OpenMP
-------------
-
-Since scikit-learn can be built without OpenMP, it's necessary to protect each
-direct call to OpenMP.
-
-The `_openmp_helpers` module, available in
-`sklearn/utils/_openmp_helpers.pyx <https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/utils/_openmp_helpers.pyx>`_
-provides protected versions of the OpenMP routines. To use OpenMP routines, they
-must be cimported from this module and not from the OpenMP library directly::
-
-  from sklearn.utils._openmp_helpers cimport omp_get_max_threads
-  max_threads = omp_get_max_threads()
-
-.. note::
-
-   The parallel loop, `prange`, is already protected by cython and can be used directly
-   from `cython.parallel`.
```

Comment:
Note for reviewers: this has been moved to `doc/developers/cython.rst` with slight modifications.